src/Microsoft.VisualStudio.Composition/LazyMetadataWrapper.cs

```diff
@@ -195,6 +197,32 @@ public bool Remove(KeyValuePair<string, object> item)
             return metadata;
         }
 
+        internal static bool TryGetMetadataValueTypeRef(IReadOnlyDictionary<string, object> metadata, string key, Resolver resolver, out object typeRef)
+        {
+            typeRef = null;
+
+            if (metadata is LazyMetadataWrapper lazyMetadata)
+            {
+                metadata = lazyMetadata.underlyingMetadata;
+            }
+
```

Comment:
I'm still a bit puzzled as to why you're unwrapping this without regard to the direction the wrapper was going.

Reply:
I'll add the check. The reason it hasn't been relevant yet is because the only time we construct a `LazyMetadataWrapper` in the `ToSubstitutedValue` direction is during serialization.